src/EditorFeatures/CSharpTest/EditAndContinue/StatementMatchingTests.cs

```diff
@@ -863,6 +863,323 @@ public void LocalFunctionDefinitions()
             expected.AssertEqual(actual);
         }
 
+        [Fact]
+        public void LocalFunctions1()
+        {
+            var src1 = "object x(object a) => a; F(x);";
+            var src2 = "F(a => a);";
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "object x(object a) => a;", "a => a" },
+                { "F(x);", "F(a => a);" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void LocalFunctions2a()
+        {
+            var src1 = @"
+F(x => x + 1, 1, y => y + 1, delegate(int x) { return x; }, async u => u);
+";
+            var src2 = @"
+int localF1(int y) => y + 1;
+int localF2(int x) => x + 1;
+int localF3(int x) => x;
+int localF4(int u) => u;
+async int localF5(int u, int v) =>  u + v;
+F(localF1, localF2, G(), localF2, localF3, localF4, localF5);
+";
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "F(x => x + 1, 1, y => y + 1, delegate(int x) { return x; }, async u => u);", "F(localF1, localF2, G(), localF2, localF3, localF4, localF5);" },
+                { "x => x + 1", "int localF2(int x) => x + 1;" },
+                { "y => y + 1", "int localF1(int y) => y + 1;" },
+                { "delegate(int x) { return x; }", "int localF3(int x) => x;" },
+                { "async u => u", "int localF4(int u) => u;" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void LocalFunctions2b()
+        {
+            var src1 = @"
+F(delegate { return x; });
+";
+            var src2 = @"
+int localF() => x;
+F(localF);
+";
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "F(delegate { return x; });", "F(localF);" },
+                { "delegate { return x; }", "int localF() => x;" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void LocalFunctions3()
+        {
+            var src1 = @"
+a += async u => u;
+";
+            var src2 = @"
+object localF(object u) => u;
+a += localF;
+";
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "a += async u => u;", "a += localF;" },
+                { "async u => u", "object localF(object u) => u;" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void LocalFunctions4()
```

Comment:
>public void LocalFunctions4() [](start = 8, length = 29)

I don't think this one is necessary. The original test validates matching of query clauses. There is nothing specific to local functions.